analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/RestrictDeserializedTypes.CSharp9.cs

```diff
@@ -78,11 +78,11 @@ public partial class Partial
 {
     public partial void Method(MemoryStream ms)
     {
-        new BinaryFormatter().Deserialize(ms);                                      // FIXME Non-compliant
+        new BinaryFormatter().Deserialize(ms);                                      // Noncompliant
 
         var formatter = new BinaryFormatter();
         formatter.Binder = new SafeBinderPartial();
-        formatter.Deserialize(ms);
+        formatter.Deserialize(ms);                                                  // Noncompliant FP: safe binder was used
```

Comment:
This FP can be fixed if you change `IsBindToTypeMethod`: check if the method has an implementation syntax:
```cs
...
&& (syntax.Body is not null || syntax.ArrowExpressionBody() is not null)
...
```